### ISSUE TITLE
TiffWriter.write with tile generator and predictor=True corrupts the predictor

### ISSUE THREAD

**Issue by jmuhlich** [cgohlke/tifffile#185]
I tried to write out a tiff using the tile generator feature, with compression and predictor enabled, and the resulting pixel data was completely corrupted. Minimal test case:

```python
import numpy as np
import tifffile

def tiles():
    t = np.empty((16, 16), "uint8")
    # Create a light gray background with a 2x2 black square in the middle.
    t[:] = 200
    t[7:9, 7:9] = 0
    yield t

with tifffile.TiffWriter("test.tif") as tiff:
    tiff.write(
        data=tiles(),
        shape=(16, 16),
        dtype="uint8",
        tile=(16, 16),
        compression="adobe_deflate",
        predictor=True,
    )

```
Changing the `yield` to `return` to pass the data up front avoids the problem, pointing to some interaction between the generator support and the predictor as the problem. No other parameters seem to have any effect (other than disabling the predictor entirely of course).

### Results
Expected:
![expected](https://user-images.githubusercontent.com/277154/225230839-d99eaa63-4a03-42ba-b4d3-0d7de7c16f64.png)

Observed: 
![observed](https://user-images.githubusercontent.com/277154/225230856-cb81d7d2-5ced-4860-86f6-64b14efb6414.png)

**Comment by cgohlke** [comment]
Thanks for reporting the bug!

The issue is that the prediction is done along the wrong axis (-2 instead of -1) because the tile is not in "normalized" shape. This should also affect PackBits compression.

To work around the issue in the current version of tifffile, yield tiles of `shape (16, 16, 1)`.

The issue should be fixed in the next version of tifffile.

**Comment by cgohlke** [comment]
Fixed in tifffile v2023.3.15.
